modules/lib/types.nix

```diff
@@ -0,0 +1,22 @@
+{ lib }:
+
+with lib;
+{
+  selectorFunction = mkOptionType {
+    name = "selectorFunction";
+    description =
+      "Function that takes an attribute set and returns a list " +
+      "containing a selection of the values of the input set";
+    check = select: isFunction select;
+    merge = _loc: defs:
+      as: concatMap (select: select as) (getValues defs);
+  };
+
+  overlayFunction = mkOptionType {
+    name = "overlayFunction";
+    description = "An overlay function, takes self and super and returns an attribute set overriding the desired attributes.";
```

Comment:
Overlong line. Better break it like in `selectFunction`. Ideally lines should be limited to 80 characters.